app/org/maproulette/data/DataManager.scala

```diff
@@ -94,6 +94,44 @@ class DataManager @Inject()(config: Config, db:Database)(implicit application:Ap
       .as(get[Option[Int]]("count").single).getOrElse(0)
   }
 
+  /**
+   * Returns the SQL to sum a user's status actions for ranking purposes
+   **/
+  private def scoreSumSQL(statusActionsTableName:String="sa") : String = {
+    s"""SUM(CASE ${statusActionsTableName}.status
+             WHEN ${Task.STATUS_FIXED} THEN ${config.taskScoreFixed}
+             WHEN ${Task.STATUS_FALSE_POSITIVE} THEN ${config.taskScoreFalsePositive}
+             WHEN ${Task.STATUS_ALREADY_FIXED} THEN ${config.taskScoreAlreadyFixed}
+             WHEN ${Task.STATUS_TOO_HARD} THEN ${config.taskScoreTooHard}
+             WHEN ${Task.STATUS_SKIPPED} THEN ${config.taskScoreSkipped}
+             ELSE 0
+           END)"""
+  }
+
+
+  /**
+   * Returns the SQL to fetch the ordered leaderboard data with rankings. Can be filtered
+   * by a list of projects, challenges, users and start/end date.
+   **/
+  private def leaderboardWithRankSQL(userFilter:Option[List[Long]]=None, projectFilter:Option[List[Long]]=None,
+                         challengeFilter:Option[List[Long]]=None, start:Option[DateTime]=None,
+                         end:Option[DateTime]=None) : String = {
+    s"""
+        SELECT users.id, users.name, users.avatar_url, ${this.scoreSumSQL()} AS score,
```

Comment:
What kind of performance do we get with this query? Considering that the status_actions table can get pretty large.

Reply:
I've added a new commit that includes a job that will generate new tables to improve performance (default cases only -- past month, 3 months, 6 months, year and no project/challenge/user filters). When this is deployed to production the conf file will need to include the interval for which this job should run. The job takes around 2.4 seconds to run on my average machine over 430k rows.

challengesLeaderboard.interval = "20 minutes"